File: examples/docs-code-snippets/src/main/java/examples/java/interactions/ShipOrder.java
package examples.java.interactions;

import com.ing.baker.recipe.annotations.FiresEvent;
import com.ing.baker.recipe.annotations.RequiresIngredient;
import com.ing.baker.recipe.javadsl.Interaction;
import examples.java.ingredients.Address;

public interface ShipOrder extends Interaction {

    record OrderShipped() {
    }

    @FiresEvent(oneOf = {OrderShipped.class})
    OrderShipped apply(@RequiresIngredient("orderId") String orderId,
                       @RequiresIngredient("address") Address address
    );
}

File: examples/docs-code-snippets/src/main/scala/examples/scala/interactions/ShipOrder.scala
package examples.scala.interactions

import com.ing.baker.recipe.scaladsl.{Event, Ingredient, Interaction}
import examples.scala.ingredients.Address

object ShipOrder {

  case class OrderShipped()

  val interaction: Interaction = Interaction(
    name = "ShipOrder",
    inputIngredients = Seq(
      Ingredient[String](name = "orderId"),
      Ingredient[Address](name = "address")
    ),
    output = Seq(
      Event[OrderShipped]
    )
  )
}
